src/Compiler/Microsoft.AspNetCore.Razor.Language/test/Syntax/FindTokenTests.cs

```diff
@@ -0,0 +1,883 @@
+    {
+        var text = """
+        @$$* Comment *@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentStar;[*];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_03_IgnoreWhitespace()
+    {
+        var text = """
+        @*$$ Comment *@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentLiteral;[ Comment ];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_03_IncludeWhitespace()
+    {
+        var text = """
+        @*$$ Comment *@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position, includeWhitespace: true);
+
+        AssertEx.Equal("""RazorCommentLiteral;[ Comment ];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_04()
+    {
+        var text = """
+        @* $$Comment *@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentLiteral;[ Comment ];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_05()
+    {
+        var text = """
+        @* Comment$$ *@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentLiteral;[ Comment ];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_06()
+    {
+        var text = """
+        @* Comment $$*@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentStar;[*];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_07()
+    {
+        var text = """
+        @* Comment *$$@
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentTransition;[@];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void RazorComment_08()
+    {
+        var text = """
+        @* Comment *@$$
+        <div />
+        """;
+        var (tree, position) = ParseWithPosition(text);
+
+        var token = tree.Root.FindToken(position);
+
+        AssertEx.Equal("""RazorCommentTransition;[@];""", SyntaxSerializer.Serialize(token).Trim());
+    }
+
+    [Fact]
+    public void OutOfRange_NoRangeInDocument()
+    {
+        var text = """
+        <div />$$
```

Comment:
Why is this out of range, but the `<div></div>$$` test isn't?

Reply:
I add 1 to this position when I request it.